test/ngAnimate/ngSwapSpec.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+describe("ngSwap", function() {
+
+  beforeEach(module('ngAnimate'));
+  beforeEach(module('ngAnimateMock'));
+
+  var element;
+  afterEach(inject(function($$jqLite) {
+    dealoc(element);
+  }));
+
+  it('should render a new container when the expression changes', inject(function($rootScope, $compile, $animate) {
+    $animate.enabled(false);
+
+    element = $compile('<div><div ng-swap="exp">{{ exp }}</div></div>')($rootScope);
+    $rootScope.$digest();
+
+    var first = element.find('div')[0];
+    expect(first).toBeFalsy();
+
+    $rootScope.exp = 'yes';
+    $rootScope.$digest();
+
+    var second = element.find('div')[0];
+    expect(second.innerText).toBe('yes');
```

Comment:
I think `textContent` is preferred over `innerText`.
<sub>(Not that it makes a real difference in tests, but `innerText` used to lead to layout thrashing.)</sub>